macros/src/tests.rs

```diff
@@ -0,0 +1,68 @@
+#[cfg(test)]
+mod tests {
+    /// Test that the macro correctly detects template parameters
+    #[test]
+    fn test_macro_detects_template_parameters() {
+        // This is a compile-time test - if the macro compiles, it works
+        let tokens = quote::quote! {
+            #[require_permissions(["relayers:get:{relayer_id}"])]
```

Comment:
can it work with multiple permissions

For example required permissions are to get specific relayer or to be able to get any relayer?

Reply:
It works with multiple ones, but in this specific case [`relayers:get:{relayer_id}`](relayers:get:{relayer_id}`) should match both a specific or all relayers (using wildcards).
- If API key contains `relayers:get:abc-abc-abc-abc` permission, then it passes. 
- If API key contains `relayers:get:*` permission, then also passes.

I think there might be endpoints where we have more than one as you said, e.g. if you want to enforce both `signing:execute:{relayer_id}` + `transactions:execute:{relayer_id}` in the same endpoint:
```rs
#[require_permissions(["signing:execute:{relayer_id}", "transactions:execute:{relayer_id}"])]
```
should work